File: sekigae.rb
#!/usr/bin/env ruby
# Usage: cat names.txt | ruby sekigae.rb

readlines.map(&:chomp).shuffle.each_slice(2).map do |p|
  len = p.map(&:size).max; p.map {|n| n.ljust(len) }
end.transpose.each do |l|
  puts l.join(' ')
end

File: sekigae.r
#!/usr/bin/Rscript
# Usage: cat names.txt | Rscript sekigae.r

frame <- read.table('stdin', stringsAsFactors = F)

names <- frame[, 1][sample(1:6, size = 6)]

dim(names) = c(2, 3)
rownames(names) <- c("front", "back")
colnames(names) <- c("1", "2", "3")

print(names)

